src/pages/iou/steps/MoneyRequestAmountPage.js

```diff
@@ -73,342 +72,305 @@ const defaultProps = {
+        const shouldReset = props.iou.id !== moneyRequestId;
+        if (shouldReset) {
+            IOU.resetMoneyRequestInfo(moneyRequestId);
+        }
 
-    /**
-     * @param {String} amount
-     * @returns {Number}
-     */
-    calculateAmountLength(amount) {
-        const leadingZeroes = amount.match(/^0+/);
-        const leadingZeroesLength = lodashGet(leadingZeroes, '[0].length', 0);
-        const absAmount = parseFloat((this.stripCommaFromAmount(amount) * 100).toFixed(2)).toString();
-
-        // The following logic will prevent users from pasting an amount that is excessively long in length,
-        // which would result in the 'absAmount' value being expressed in scientific notation or becoming infinity.
-        if (/\D/.test(absAmount)) {
-            return CONST.IOU.AMOUNT_MAX_LENGTH + 1;
+        if (_.isEmpty(props.iou.participants) || props.iou.amount === 0 || shouldReset) {
+            Navigation.goBack(ROUTES.getMoneyRequestRoute(iouType.current, reportID.current), true);
         }
 
-        /*
-        Return the sum of leading zeroes length and absolute amount length(including fraction digits).
-        When the absolute amount is 0, add 2 to the leading zeroes length to represent fraction digits.
-        */
-        return leadingZeroesLength + (absAmount === '0' ? 2 : absAmount.length);
-    }
+        return () => {
+            prevMoneyRequestId.current = props.iou.id;
+        };
+    }, [props.iou.participants, props.iou.amount, props.iou.id]);
 
-    /**
-     * Check if amount is a decimal up to 3 digits
-     *
-     * @param {String} amount
-     * @returns {Boolean}
-     */
-    validateAmount(amount) {
-        const decimalNumberRegex = new RegExp(/^\d+(,\d+)*(\.\d{0,2})?$/, 'i');
-        return amount === '' || (decimalNumberRegex.test(amount) && this.calculateAmountLength(amount) <= CONST.IOU.AMOUNT_MAX_LENGTH);
-    }
+    useEffect(() => {
+        const currencyParam = lodashGet(props.route.params, 'currency', '');
+        const prevCurrencyParam = prevPropCurrencyParam.current;
+        if (currencyParam !== '' && prevCurrencyParam !== currencyParam) {
+            setSelectedCurrencyCode(currencyParam);
+        }
 
-    /**
-     * Strip comma from the amount
-     *
-     * @param {String} amount
-     * @returns {String}
-     */
-    stripCommaFromAmount(amount) {
-        return amount.replace(/,/g, '');
-    }
+        if (prevPropIOUCurrency.current !== props.iou.currency) {
+            setSelectedCurrencyCode(props.iou.currency);
+        }
 
-    /**
-     * Strip spaces from the amount
-     *
-     * @param {String} amount
-     * @returns {String}
-     */
-    stripSpacesFromAmount(amount) {
-        return amount.replace(/\s+/g, '');
-    }
+        return () => {
+            prevPropIOUCurrency.current = props.iou.currency;
+            prevPropCurrencyParam.current = lodashGet(props.route, 'params.currency');
+        };
+    }, [props.route, props.iou.currency]);
```

Comment:
Replace this `useEffect` with 2 simpler ones and remove the two variables `prevPropIOUCurrency` and `prevPropCurrencyParam`.


```js
    useEffect(() => {
        if (!props.route.params.currency) {
            return;
        }

        setSelectedCurrencyCode(props.route.params.currency);
    }, [props.route.params.currency]);

    useEffect(() => {
        setSelectedCurrencyCode(props.iou.currency);
    }, [props.iou.currency]);
```